File: Simulation/randomize_binary.R
### Function for randomizing binary 0s and 2s so 0s are not assumed to be ancestral ###

randomize_binary <- function(binary_allele, randomallele){
  
  new_seq <- list()
  
  if (binary_allele == "0"){
    new_seq <- append(new_seq, randomallele, after = length(new_seq))
  } else if (binary_allele == "2" && randomallele == "2"){
    new_seq <- append(new_seq, "0", after = length(new_seq))
  } else if (binary_allele == "2" && randomallele == "0"){
    new_seq <- append(new_seq, "2", after = length(new_seq))
  } else if (binary_allele == "1"){
    new_seq <- append(new_seq, "1", after = length(new_seq))
  } else {
    print("Randomization Error")
  }
  
  return(new_seq)
  
}

File: Simulation/randomize_binary.py
##### Script for randomizing designation of binary 0s and 2s #####
##### Note: randomstart is the list to convert 0s to #####

def randomize_binary(seq, randomstart):

	for n in range(0, len(seq), 1):
		if seq[n] == '0':
			seq[n] == randomstart[n]
		elif seq[n] == '2' and randomstart[n] == '2':
			seq[n] == '0'
		elif seq[n] == '2' and randomstart[n] == '0':
			seq[n] == '2'
		elif seq[n] == '1':
			continue
		else:
			print('Randomization Error' + '\t' + seq[n])
			
		return(seq)
